src/controllers/movie.js

```diff
@@ -0,0 +1,159 @@
+const {
+  PrismaClientKnownRequestError,
+  PrismaClientValidationError,
+} = require("@prisma/client/runtime/library");
+const {
+  getMoviesDb,
+  getMoviesWhereAndDb,
+  getMoviesWhereLtDb,
+  getMoviesWhereGtDb,
+  createMovieDb,
+  createMovieWithScreeningsDb,
+  getMovieByIdDb,
+  updateMovieDb,
+  getMovieByTitleDb,
+  updateMovieWithScreeningsDb,
+} = require("../domains/movie");
+
+const getMovies = async (req, res) => {
+  let movies;
+  const { runtimeLt, runtimeGt } = req.query;
+  if (runtimeLt && runtimeGt) {
+    const runtimeLtNum = Number(runtimeLt);
+    const runtimeGtNum = Number(runtimeGt);
+    movies = await getMoviesWhereAndDb(runtimeGtNum, runtimeLtNum);
+  }
+
+  if (runtimeLt && !runtimeGt) {
+    const runtimeLtNum = Number(runtimeLt);
+    movies = await getMoviesWhereLtDb(runtimeLtNum);
+  }
+
+  if (runtimeGt && !runtimeLt) {
+    const runtimeGtNum = Number(runtimeGt);
+    movies = await getMoviesWhereGtDb(runtimeGtNum);
+  }
+
+  if (!runtimeLt && !runtimeGt) {
+    movies = await getMoviesDb();
+  }
+
+  res.json({ movies: movies });
+};
+
+const createMovie = async (req, res) => {
+  const { title, runtimeMins, screenings } = req.body;
+  let createdMovie;
+
+  if (!screenings || screenings.length === 0) {
+    try {
+      createdMovie = await createMovieDb(title, runtimeMins);
+    } catch (e) {
+      if (e instanceof PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+          res
+            .status(409)
+            .json({ error: "A movie with the provided title already exists" });
+          return;
+        }
+      }
+      if (e instanceof PrismaClientValidationError) {
+        // e.code is undefined for P1012 ("Arguement {} is missing.")
+        // hence testing that the ends of the message matches what
+        // we'd expect instead.
+        if (e.message.endsWith("is missing.")) {
+          res.status(400).json({ error: "Missing fiels in request body" });
+          return;
+        }
+      }
+    }
+  }
+
+  if (screenings && screenings.length !== 0) {
+    try {
+      createdMovie = await createMovieWithScreeningsDb(
+        title,
+        runtimeMins,
+        screenings
+      );
+    } catch (e) {
+      if (e instanceof PrismaClientKnownRequestError) {
+        if (e.code === "P2002") {
+          res
+            .status(409)
+            .json({ error: "A movie with the provided title already exists" });
+          return;
+        }
+      }
+    }
+  }
+  res.status(201).json({ movie: createdMovie });
+};
+
+const getMovieBy = async (req, res) => {
+  const { searchparam } = req.params;
+  let movie;
+
+  if (Number(searchparam)) {
+    const idNum = Number(searchparam);
+    movie = await getMovieByIdDb(idNum);
+  } else {
+    [movie] = await getMovieByTitleDb(searchparam);
```

Comment:
What's with the square brackets here?

Reply:
the issue is, getMovieByTitleDb() relies on a .findMany() instead of a findUnique() (I'll be honest, I do remember the findUnique() failing, but I can't remember why, or why I just went with findMany() instead.)

With  getMovieByTitleDb returning an array, I ended up destructuring it on line 101, so that the `movie` variable would be of the same type no matter which function it gets its value from. At least, that was the intention - I probably should have just used findUnique() in both cases, though